### DISCUSSION TITLE
Weblate stores Arabic under the ar-AA tag, which is not BCP47 compliant

### DISCUSSION THREAD

**Discussion by steam3d** [WeblateOrg/weblate#13299] [General]
I think weblate is not storing Arabic correctly using the ar-AA tag. The only tags in the BCP47 list are `ar-AE, ar-BH, ar-DZ, ar-EG, ar-IQ, ar-JO, ar-KW, ar-LB, ar-LY, ar-MA, ar-OM, ar-QA, ar-SA, ar-SD, ar-SY, ar-TN, ar-YE`

**Comment by nijel** [comment]
Weblate does not create such file (see https://github.com/WeblateOrg/language-data/blob/3d4dd83374f4e9091ab5dae4a106f4d730acf1d8/languages.csv#L26 for list of languages defined in Weblate).

**Comment by steam3d** [comment]
@nijel  I have this setting on. In my repository weblate created the folder ar-AA. 

![image](https://github.com/user-attachments/assets/18ba9e42-8924-42f9-a617-74325530baee)

**Reply by nijel** [reply]
Ah, you will indeed end up with `ar-AA` this way. The question is what should be the correct behavior here, as binding Arabic to a single country seems wrong as well (`ar-AA` stands for Arabic, Arabic Countries).

**Reply by steam3d** [reply]
@nijel Maybe I'm missing something, but I didn't find any mention of ar-AA in BCP. Most likely, in this case, simply `ar` would be the correct option. Because ar-AA can lead to unexpected errors in the application, since such a locale is not correct, at least in Windows applications.

In any case, are there any recommendations on how I can rename ar-AA to ar?
Will weblate be able to rebuild the repository correctly?

**Reply by nijel** [reply]
If you don't need country code choose *BCP style using hyphen as a separator*.

The *BCP style using hyphen as a separator, including country code* is there specifically for cases where country code is needed. In that case, I still don't see a better option than using an invalid code.

**Reply by steam3d** [reply]
@nijel No, I just need the “en-US” format, but I would like to correct “ar-AA” into a valid BCP tag “ar”. Is this possible to do in the current weblate? Or should I manually rename and update the rows every time?

**Reply by nijel** [reply]
There is no configuration which would turn `en` into `en-US` and not `ar` into `ar-AA`. You can have either both or none. If you want `en-US` just choose *English (United States)* as language.

Furthermore, this setting takes effect only when creating languages, so if you have very specific requirements on this, just create languages manually.
